### ISSUE TITLE
Scan fails with undefined method `range' for nil:NilClass

### ISSUE THREAD

**Issue by mvz** [castwide/solargraph#503]
The following minimal code will trigger an error 'undefined method \`range' for nil:NilClass' when running `solargraph scan`:

```ruby
Foo.class_eval do
  foo = :bar
end
```

This error occurs with solargraph 0.44.1. I've tested on Ruby 3.0.2 and 2.7.4, and the error occurs on both versions.

Backtrace on Ruby 3.0.2:

```
/home/matijs/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/solargraph-0.44.1/lib/solargraph/parser/rubyvm/node_processors/lvasgn_node.rb:11:in `process': undefined method `range' for nil:NilClass (NoMethodError)
	from /home/matijs/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/solargraph-0.44.1/lib/solargraph/parser/node_processor.rb:38:in `process'
	from /home/matijs/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/solargraph-0.44.1/lib/solargraph/parser/node_processor/base.rb:46:in `block in process_children'
	from /home/matijs/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/solargraph-0.44.1/lib/solargraph/parser/node_processor/base.rb:44:in `each'
	from /home/matijs/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/solargraph-0.44.1/lib/solargraph/parser/node_processor/base.rb:44:in `process_children'
	from /home/matijs/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/solargraph-0.44.1/lib/solargraph/parser/rubyvm/node_processors/scope_node.rb:9:in `process'
	from /home/matijs/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/solargraph-0.44.1/lib/solargraph/parser/node_processor.rb:38:in `process'
	from /home/matijs/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/solargraph-0.44.1/lib/solargraph/parser/node_processor/base.rb:46:in `block in process_children'
	from /home/matijs/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/solargraph-0.44.1/lib/solargraph/parser/node_processor/base.rb:44:in `each'
	from /home/matijs/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/solargraph-0.44.1/lib/solargraph/parser/node_processor/base.rb:44:in `process_children'
	from /home/matijs/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/solargraph-0.44.1/lib/solargraph/parser/rubyvm/node_processors/block_node.rb:17:in `process'
	from /home/matijs/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/solargraph-0.44.1/lib/solargraph/parser/node_processor.rb:38:in `process'
	from /home/matijs/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/solargraph-0.44.1/lib/solargraph/parser/rubyvm/class_methods.rb:29:in `map'
	from /home/matijs/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/solargraph-0.44.1/lib/solargraph/source_map/mapper.rb:26:in `map'
	from /home/matijs/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/solargraph-0.44.1/lib/solargraph/source_map/mapper.rb:50:in `map'
	from /home/matijs/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/solargraph-0.44.1/lib/solargraph/source_map.rb:157:in `map'
	from /home/matijs/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/solargraph-0.44.1/lib/solargraph/library.rb:455:in `block in map!'
	from /home/matijs/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/solargraph-0.44.1/lib/solargraph/library.rb:454:in `each'
	from /home/matijs/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/solargraph-0.44.1/lib/solargraph/library.rb:454:in `map!'
	from /home/matijs/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/solargraph-0.44.1/lib/solargraph/api_map.rb:119:in `load'
	from /home/matijs/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/solargraph-0.44.1/lib/solargraph/shell.rb:177:in `block in scan'
	from /home/matijs/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/benchmark-0.2.0/lib/benchmark.rb:296:in `measure'
	from /home/matijs/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/solargraph-0.44.1/lib/solargraph/shell.rb:176:in `scan'
	from /home/matijs/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/thor-1.1.0/lib/thor/command.rb:27:in `run'
	from /home/matijs/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/thor-1.1.0/lib/thor/invocation.rb:127:in `invoke_command'
	from /home/matijs/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/thor-1.1.0/lib/thor.rb:392:in `dispatch'
	from /home/matijs/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/thor-1.1.0/lib/thor/base.rb:485:in `start'
	from /home/matijs/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/solargraph-0.44.1/bin/solargraph:5:in `<top (required)>'
	from /home/matijs/.rbenv/versions/3.0.2/bin/solargraph:23:in `load'
	from /home/matijs/.rbenv/versions/3.0.2/bin/solargraph:23:in `<main>'
```

**Comment by castwide** [comment]
Confirmed. I'll work on a fix for a patch release.

**Comment by castwide** [comment]
Fixed in v0.44.2.

**Comment by mvz** [comment]
Thanks, @castwide, I can confirm that the error no longer happens.
